Trouble installing on WSL

I am submitting a **documentation issue**.

**The files in question are:**
readthedocs pages

**The problem is in the following category/categories:**
Undocumented Information

**Description of the problem:**
Currently, the docs do not make it clear what setup is required for installation on WSL. This leads to issues with the installation that occur at the `make` step. Currently, these issues appear to be tied to how ROOT is installed.

Here is what I know so far:
* Installing ROOT v6.24/00 via `nix-env` or v6.24/06 on Ubuntu Bionic Beaver via pre-compiled binary will cause `make` to fail.
* Installing ROOT v6.22/00 or v6.24/00 via `conda` on Ubuntu Bionic Beaver allows `make` to succeed.
* [added] Installing ROOT v6.24/06 by building from source on Ubuntu Bionic Beaver allows `make` to succeed.
* [added] Attempting to install ROOT v6.24/06 and v6.20/00 from pre-compiled binaries caused `make` to fail on Ubuntu Bionic Beaver.

Other possible issues/explanations:
* The code does not make in NixOS on WSL.
* The code does not make in NixOS at all.
* ~~The code does not make using ROOT v6.24 on WSL.~~

Things that would be helpful to determine:
* ~~Does the code make if ROOT is compiled from source? (Currently testing)~~
* ~~Does the code still make in v6.22/00 if ROOT is installed from something other than conda?~~
* ~~Does the code make in v6.24 if ROOT is installed via conda?~~
* Does the code make in a "pure" NixOS environment, with ROOT installed via nix-env? (I don't know how we would test this, but it would be good to check!)

Tagging @altavir to continue our discussion from the review thread.

The error is the following: /usr/bin/ld: /nix/store/7g0k3gjnvswvllqjbcr02vsydzgpgzgm-xz-5.2.5/lib/liblzma.so.5: undefined reference to `pthread_sigmask@GLIBC_2.32'. It seems like a glbc conflict inside root.

Removing Root dependency should indeed fix a lot of issues. You use it from python scipts anyway. It does not seem to be critical for the simulation iteslf.

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,71 +4,82 @@ jobs:
     # more recent build envs:  https://changelog.travis-ci.com/
     - os: osx
       osx_image: xcode11.4 
-      env:  ROOTDIST="root_v6.20.00.macosx64-10.15-clang110.tar.gz"; BUILDLINE=2
+      env:  ROOTDIST="root_v6.20.00.macosx64-10.15-clang110.tar.gz"
       language: generic 
       python: 3.7
     - os: osx
       osx_image: xcode12 
-      env:  ROOTDIST="root_v6.24.06.macos-10.15-x86_64-clang120.tar.gz"; BUILDLINE=472
+      env:  ROOTDIST="root_v6.24.06.macos-10.15-x86_64-clang120.tar.gz"
       language: generic 
       python: 3.7
     - os: osx
       osx_image: xcode13.1 
-      env:  ROOTDIST="root_v6.24.06.macos-11.5-x86_64-clang120.tar.gz"; BUILDLINE=539
+      env:  ROOTDIST="root_v6.24.06.macos-11.5-x86_64-clang120.tar.gz"
       language: generic 
       python: 3.7
     - os: linux
       dist: xenial
-      env:  ROOTDIST="root_v6.20.00.Linux-ubuntu16-x86_64-gcc5.4.tar.gz"; BUILDLINE=203
+      env:  ROOTDIST="root_v6.20.00.Linux-ubuntu16-x86_64-gcc5.4.tar.gz"
       language: python
       python: 3.7
     - os: linux
       dist: bionic
-      env:  ROOTDIST="root_v6.20.00.Linux-ubuntu18-x86_64-gcc7.4.tar.gz"; BUILDLINE=270
+      env:  ROOTDIST="root_v6.20.00.Linux-ubuntu18-x86_64-gcc7.4.tar.gz"
       language: python
       python: 3.7
     - os: linux
       dist: xenial
-      env:  ROOTDIST="root_v6.24.06.Linux-ubuntu16-x86_64-gcc5.4.tar.gz"; BUILDLINE=606
+      env:  ROOTDIST="root_v6.24.06.Linux-ubuntu16-x86_64-gcc5.4.tar.gz"
       language: python
       python: 3.7
     - os: linux
       dist: bionic
-      env:  ROOTDIST="root_v6.24.06.Linux-ubuntu18-x86_64-gcc7.5.tar.gz"; BUILDLINE=673
+      env:  ROOTDIST="root_v6.24.06.Linux-ubuntu18-x86_64-gcc7.5.tar.gz"
       language: python
       python: 3.7
     - os: linux
       dist: focal 
-      env:  ROOTDIST="root_v6.24.06.Linux-ubuntu20-x86_64-gcc9.3.tar.gz"; BUILDLINE=740
+      env:  ROOTDIST="root_v6.24.06.Linux-ubuntu20-x86_64-gcc9.3.tar.gz"
+      language: python
+      python: 3.7
+    - os: windows
       language: python
       python: 3.7
 
 install:
-   - sudo chmod 777 .travis/install.sh
-   - ls -lh .travis/install.sh
-   - source .travis/install.sh
+  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then curl -L -o ubuntu-1804.zip https://aka.ms/wsl-ubuntu-1804; fi
+  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then powershell.exe -NoProfile -Command "Expand-Archive -Path ubuntu-1804.zip -DestinationPath Ubuntu"; fi
+  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then Ubuntu/ubuntu1804.exe install --root; fi
+  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then wsl.exe apt update; fi
+  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then wsl.exe apt install -y python3-pip; fi
+  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then wsl; fi
+  - source activate piptest
+  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then conda install -c conda-forge root
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; echo "installing..."; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; echo $ROOTDIST; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; echo "wget command..."; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; echo "wget https://root.cern/download/"$ROOTDIST; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then wget https://root.cern/download/$ROOTDIST; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then tar -xzf $ROOTDIST; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then source root/bin/thisroot.sh; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; echo "root libraries.."; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; echo `root-config --cflags --glibs`; fi
+  - root --version
+  - sudo chmod 777 .travis/install.sh
+  - ls -lh .travis/install.sh
+  - source .travis/install.sh
 
-   - echo "installing..."
-   - echo $ROOTDIST
-   - echo "wget command..." 
-   - echo "wget https://root.cern/download/"$ROOTDIST 
-   - wget https://root.cern/download/$ROOTDIST 
-   - tar -xzf $ROOTDIST 
-   - source root/bin/thisroot.sh
-   - echo "root libraries.."
-   - echo `root-config --cflags --glibs`
+  # install and set up lcov
+  - wget http://downloads.sourceforge.net/ltp/lcov-1.14.tar.gz
+  - tar -zxvf lcov-1.14.tar.gz
+  - cd lcov-1.14
+  - sudo make install
+  - cd ../
+  - lcov --directory ./ --zerocounters
 
-   # install and set up lcov
-   - wget http://downloads.sourceforge.net/ltp/lcov-1.14.tar.gz
-   - tar -zxvf lcov-1.14.tar.gz
-   - cd lcov-1.14
-   - sudo make install
-   - cd ../
-   - lcov --directory ./ --zerocounters
-
-   # give self permission to run scripts
-   - sudo chmod 777 tests/test_realizeCascades.sh
-   - sudo chmod 777 get_checksums.sh
+  # give self permission to run scripts
+  - sudo chmod 777 tests/test_realizeCascades.sh
+  - sudo chmod 777 get_checksums.sh
 
 script:
   #make repository code (my code)

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,82 +4,71 @@ jobs:
     # more recent build envs:  https://changelog.travis-ci.com/
     - os: osx
       osx_image: xcode11.4 
-      env:  ROOTDIST="root_v6.20.00.macosx64-10.15-clang110.tar.gz"
+      env:  ROOTDIST="root_v6.20.00.macosx64-10.15-clang110.tar.gz"; BUILDLINE=2
       language: generic 
       python: 3.7
     - os: osx
       osx_image: xcode12 
-      env:  ROOTDIST="root_v6.24.06.macos-10.15-x86_64-clang120.tar.gz"
+      env:  ROOTDIST="root_v6.24.06.macos-10.15-x86_64-clang120.tar.gz"; BUILDLINE=472
       language: generic 
       python: 3.7
     - os: osx
       osx_image: xcode13.1 
-      env:  ROOTDIST="root_v6.24.06.macos-11.5-x86_64-clang120.tar.gz"
+      env:  ROOTDIST="root_v6.24.06.macos-11.5-x86_64-clang120.tar.gz"; BUILDLINE=539
       language: generic 
       python: 3.7
     - os: linux
       dist: xenial
-      env:  ROOTDIST="root_v6.20.00.Linux-ubuntu16-x86_64-gcc5.4.tar.gz"
+      env:  ROOTDIST="root_v6.20.00.Linux-ubuntu16-x86_64-gcc5.4.tar.gz"; BUILDLINE=203
       language: python
       python: 3.7
     - os: linux
       dist: bionic
-      env:  ROOTDIST="root_v6.20.00.Linux-ubuntu18-x86_64-gcc7.4.tar.gz"
+      env:  ROOTDIST="root_v6.20.00.Linux-ubuntu18-x86_64-gcc7.4.tar.gz"; BUILDLINE=270
       language: python
       python: 3.7
     - os: linux
       dist: xenial
-      env:  ROOTDIST="root_v6.24.06.Linux-ubuntu16-x86_64-gcc5.4.tar.gz"
+      env:  ROOTDIST="root_v6.24.06.Linux-ubuntu16-x86_64-gcc5.4.tar.gz"; BUILDLINE=606
       language: python
       python: 3.7
     - os: linux
       dist: bionic
-      env:  ROOTDIST="root_v6.24.06.Linux-ubuntu18-x86_64-gcc7.5.tar.gz"
+      env:  ROOTDIST="root_v6.24.06.Linux-ubuntu18-x86_64-gcc7.5.tar.gz"; BUILDLINE=673
       language: python
       python: 3.7
     - os: linux
       dist: focal 
-      env:  ROOTDIST="root_v6.24.06.Linux-ubuntu20-x86_64-gcc9.3.tar.gz"
-      language: python
-      python: 3.7
-    - os: windows
+      env:  ROOTDIST="root_v6.24.06.Linux-ubuntu20-x86_64-gcc9.3.tar.gz"; BUILDLINE=740
       language: python
       python: 3.7
 
 install:
-  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then curl -L -o ubuntu-1804.zip https://aka.ms/wsl-ubuntu-1804; fi
-  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then powershell.exe -NoProfile -Command "Expand-Archive -Path ubuntu-1804.zip -DestinationPath Ubuntu"; fi
-  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then Ubuntu/ubuntu1804.exe install --root; fi
-  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then wsl.exe apt update; fi
-  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then wsl.exe apt install -y python3-pip; fi
-  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then wsl; fi
-  - source activate piptest
-  - if [ "$TRAVIS_OS_NAME" = "windows" ]; then conda install -c conda-forge root
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; echo "installing..."; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; echo $ROOTDIST; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; echo "wget command..."; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; echo "wget https://root.cern/download/"$ROOTDIST; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then wget https://root.cern/download/$ROOTDIST; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then tar -xzf $ROOTDIST; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then source root/bin/thisroot.sh; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; echo "root libraries.."; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; echo `root-config --cflags --glibs`; fi
-  - root --version
-  - sudo chmod 777 .travis/install.sh
-  - ls -lh .travis/install.sh
-  - source .travis/install.sh
+   - sudo chmod 777 .travis/install.sh
+   - ls -lh .travis/install.sh
+   - source .travis/install.sh
 
-  # install and set up lcov
-  - wget http://downloads.sourceforge.net/ltp/lcov-1.14.tar.gz
-  - tar -zxvf lcov-1.14.tar.gz
-  - cd lcov-1.14
-  - sudo make install
-  - cd ../
-  - lcov --directory ./ --zerocounters
+   - echo "installing..."
+   - echo $ROOTDIST
+   - echo "wget command..." 
+   - echo "wget https://root.cern/download/"$ROOTDIST 
+   - wget https://root.cern/download/$ROOTDIST 
+   - tar -xzf $ROOTDIST 
+   - source root/bin/thisroot.sh
+   - echo "root libraries.."
+   - echo `root-config --cflags --glibs`
 
-  # give self permission to run scripts
-  - sudo chmod 777 tests/test_realizeCascades.sh
-  - sudo chmod 777 get_checksums.sh
+   # install and set up lcov
+   - wget http://downloads.sourceforge.net/ltp/lcov-1.14.tar.gz
+   - tar -zxvf lcov-1.14.tar.gz
+   - cd lcov-1.14
+   - sudo make install
+   - cd ../
+   - lcov --directory ./ --zerocounters
+
+   # give self permission to run scripts
+   - sudo chmod 777 tests/test_realizeCascades.sh
+   - sudo chmod 777 get_checksums.sh
 
 script:
   #make repository code (my code)

--- a/RELEASENOTES.md
+++ b/RELEASENOTES.md
@@ -1,6 +1,7 @@
 ## Future Releases (v1.3.2) Date XX.XX.XX 
 
 * PR #78 updated documentation to systematicall include units of all parameters (Issue #74)
+* PR #79 updated documentation to clarify a requirement when installing ROOT dependency on WSL (Issue #76)
 
 ## Release (v1.3.1) Date 21.12.30 
 

--- a/docs/source/01_Getting_Started.rst
+++ b/docs/source/01_Getting_Started.rst
@@ -48,6 +48,10 @@ On Linux machines, you can also install `ROOT` via a `pre-packaged binary
 The one provided above is used for Focal. Bionic uses ``root_v6.24.06.Linux-ubuntu18-x86_64-gcc7.5.tar.gz``
 and Xenial uses ``root_v6.24.06.Linux-ubuntu20-x86_64-gcc9.3.tar.gz``.
 
+**If you are using WSL,** you will need to `install the conda package <https://root.cern/install#conda>`_ (recommended) 
+or `build ROOT from source <https://root.cern/install#build-from-source>`_. 
+Using a pre-compiled binary or installing from most package managers will not work for WSL users.
+
 """"""""""""""""""""""""""""""""""""""""""""""""""""""""""
 2. `gcc <https://gcc.gnu.org/>`_
 """"""""""""""""""""""""""""""""""""""""""""""""""""""""""

--- a/nrCascadeSim_withDocs.yml
+++ b/nrCascadeSim_withDocs.yml
@@ -17,4 +17,5 @@ dependencies:
     - numpy==1.19.2
     - scipy==1.6.2
     - matplotlib==3.3.2
+    - breathe
 #  - root

